examples/metrics/index.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as fetch from "node-fetch";
+
+const topic = new aws.sns.Topic("sites-to-process-topic");
+const subscription = topic.onEvent("for-each-url", async (event) => {
+    const records = event.Records || [];
+    for (const record of records) {
+        const url = record.Sns.Message;
+
+        console.log(`${url}: Processing`);
+
+        // Fetch the contents at the URL
+        console.log(`${url}: Getting`);
+        try {
+            const res = await fetch.default(url);
+        } catch (err) {
+            console.log(`${url}: Failed to GET`);
+            return;
+        }
+    }
+});
+
+const metric = aws.lambda.metricDuration({ unit: "Seconds", dimensions: { FunctionName: subscription.func.name } });
```

Comment:
Is there a way to simplify this so we create the alarm and metric at the same time? Or is that something that would go in the AWSX layer?

Reply:
it's a good question.  we could consdier just having similar things like "alarmDuration" to go along with all the metrics.  The things that are pushign me away from doing that are:

1. we'd end up with every 'metrixXXX' method duplicated.
2. we'll eventually want to tie these metrics in cloudwatch dashboards/widgets.  And then i'm loathe to then have widgetXXX methods to go along with all the metricXXX and alarmXXX methods.  

I think it's good to have 'metric' as the underlying primitive, and then provide the helpers on top of that.